logbook-core/src/main/java/org/zalando/logbook/DynamicPathFilter.java

```diff
@@ -0,0 +1,66 @@
+package org.zalando.logbook;
+
+import java.util.Objects;
+import java.util.function.UnaryOperator;
+
+final class DynamicPathFilter implements PathFilter {
+
+    private final UnaryOperator<String> replacementFunction;
+    private final String[] parts;
+
+    DynamicPathFilter(final UnaryOperator<String> replacementFunction, final String pathExpression) {
+        this.replacementFunction = replacementFunction;
+
+        final String[] parts = pathExpression.split("/");
+
+        for (int i = 0; i < parts.length; i++) {
+            final String part = parts[i];
+            if (part.startsWith("{") && part.endsWith("}")) {
+                parts[i] = null;
+            }
+        }
+
+        this.parts = parts;
+    }
+
+
+    @Override
+    public String filter(String path) {
```

Comment:
Is this logic similar to the one from the default path filter and if yes, can it be shared somehow?

Reply:
Yes, the logic is similar, but not quite. I prefer to overwrite the method instead of doing a generic one with many IFs

Reply:
Yeah, good call!